schema/resolvers.js

```diff
@@ -216,7 +218,7 @@ const Resolvers = {
       ctx.repositories.Routing.findRoutingRuleSetByQuestionPageId({
         questionPageId
       }),
-    title: (page, args) => formatRichText(page.title, args.format)
+    displayName: page => getDisplayName(page, ["title"], "Untitled Page")
```

Comment:
There is definitely a pattern it follows at the moment but thought it would be best to leave it flexible. We could maybe set a default value it calculates based on the entity type if a fallback value is not provided.  IMO this just adds extra unnecessary complexity to the method but maybe the consistency offers a better UX.

Reply:
I would say consistency is better here and worth the complexity otherwise we could quite easily fall back into our current situation where different places have different conventions.

Reply:
I would need to change the signature of this method to make this work to:

`getDisplayName(entity, keys, typename, fallbackValue )`

To use `typename` it seems ok: `getDisplayName(answer, ["alias", "title", "label"], "Answer")`
To use `fallbackValue` it seems ugly: `getDisplayName(answer, ["alias", "title", "label"], null, "Untitled Answer")`

I'm still struggling to see more value than consistency by changing this.

Reply:
Sorry, think I might have explained myself badly in the comments. The intention was to try move that fallback value and the order of priority into the util function to try and improve consistency. I was thinking something like this: 

```
const defaultNames = {
  Section: "Untitled Section",
  QuestionPage: "Untitled Page",
  Option: "Untitled Label",
  BasicAnswer: "Untitled Answer",
  MultipleChoiceAnswer: "Untitled Answer",
  CompositeAnswer: "Untitled Answer"
};

const getName = (obj, typeName) => {
  const title = find(pick(obj, ["alias", "title", "label"]), foo => foo);
  return title ? title : defaultNames[typeName];
};
```

Then in the resolvers we just need to call it with getName(entity, field), was thinking this would just enforce it a little better, plus we wouldn't have to rely on always sending in the correct order and fallback value? Does that make sense?

Reply:
@SamGodwin2  @tgandrews I still feel having to adjust the internals of a method when adding a new type is bad practice.  However, will adjust to the above.

Reply:
Maybe this is something that could be solved with an entity that could contain the config for the default value and field lookup order (if it needs to be overridden). But as it stands at the moment I think @SamGodwin2's solution is better than pushing it into the resolvers.